stripe/src/test/java/com/stripe/android/view/CardInputWidgetTest.java

```diff
@@ -548,6 +551,101 @@ public void addDinersClubCard_scrollsOver_andSetsExpectedDisplayValues() {
+        assertTrue(mCardNumberEditText.hasFocus());
+    }
+
+    @Test
+    public void setExpirationDate_withValidData_setsCorrectValues() {
+        mCardInputWidget.setExpiryDate(12, 79);
+        assertEquals("12/79", mExpiryEditText.getText().toString());
+    }
+
+    @Test
+    public void setCvcCode_withValidData_setsValue() {
+        mCardInputWidget.setCvcCode("123");
+        assertEquals("123", mCvcEditText.getText().toString());
+    }
+
+    @Test
+    public void setCvcCode_withLongString_truncatesValue() {
+        mCardInputWidget.setCvcCode("1234");
+        assertEquals("123", mCvcEditText.getText().toString());
+    }
+
+    @Test
+    public void setCvcCode_whenCardBrandIsAmericanExpress_allowsFourDigits() {
+        mCardInputWidget.setCardNumber(VALID_AMEX_NO_SPACES);
+        mCardInputWidget.setCvcCode("1234");
+        assertEquals("1234", mCvcEditText.getText().toString());
+    }
+
+    @Test
+    public void setAllCardFields_whenValidValues_allowsGetCardWithExpectedValues() {
+        if (Calendar.getInstance().get(Calendar.YEAR) > 2079) {
+            fail("Update the code with a date that is still valid. Also, hello from the past.");
+        }
+
+        mCardInputWidget.setCardNumber(VALID_AMEX_NO_SPACES);
+        mCardInputWidget.setExpiryDate(12, 2079);
+        mCardInputWidget.setCvcCode("1234");
+        Card card = mCardInputWidget.getCard();
+        assertNotNull(card);
+        assertEquals(VALID_AMEX_NO_SPACES, card.getNumber());
+        assertNotNull(card.getExpMonth());
+        assertNotNull(card.getExpYear());
+        assertEquals(12, (int) card.getExpMonth());
+        assertEquals(2079, (int) card.getExpYear());
+        assertEquals("1234", card.getCVC());
+        assertEquals(Card.AMERICAN_EXPRESS, card.getBrand());
+    }
+
+    @Test
+    public void shouldIconShowBrand_whenCvcNotFocused_isAlwaysTrue() {
+        assertTrue(shouldIconShowBrand(Card.AMERICAN_EXPRESS, false, "1234"));
+        assertTrue(shouldIconShowBrand(Card.AMERICAN_EXPRESS, false, ""));
+        assertTrue(shouldIconShowBrand(Card.VISA, false, "333"));
+        assertTrue(shouldIconShowBrand(Card.DINERS_CLUB, false, "12"));
+        assertTrue(shouldIconShowBrand(Card.DISCOVER, false, null));
+        assertTrue(shouldIconShowBrand(Card.JCB, false, "7"));
+    }
+
+    @Test
+    public void shouldIconShowBrand_whenAmexAndLengthNotFour_isFalse() {
+        assertFalse(shouldIconShowBrand(Card.AMERICAN_EXPRESS, true, ""));
+        assertFalse(shouldIconShowBrand(Card.AMERICAN_EXPRESS, true, "1"));
+        assertFalse(shouldIconShowBrand(Card.AMERICAN_EXPRESS, true, "22"));
+        assertFalse(shouldIconShowBrand(Card.AMERICAN_EXPRESS, true, "333"));
+    }
+
+    @Test
+    public void shouldIconShowBrand_whenAmexAndLengthIsFour_isTrue() {
+        assertTrue(shouldIconShowBrand(Card.AMERICAN_EXPRESS, true, "1234"));
+    }
+
+    @Test
+    public void shouldIconShowBrand_whenNotAmexAndLengthIsNotThree_isFalse() {
```

Comment:
That refers to the string length (and the ones below are all less than three characters long)